pallets/grants/src/lib.rs

```diff
@@ -377,7 +391,12 @@ impl<T: Config> Pallet<T> {
 
 		T::Currency::transfer(from, to, schedule_amount, ExistenceRequirement::AllowDeath)?;
 		T::Currency::set_lock(VESTING_LOCK_ID, to, total_amount, WithdrawReasons::all());
-		<VestingSchedules<T>>::mutate(to, |v| (*v).push(schedule));
+		<VestingSchedules<T>>::mutate(to, |vesting_schedules| -> DispatchResult {
+			vesting_schedules
+				.try_push(schedule)
+				.map_err(|_| <Error<T>>::MaxScheduleOverflow)?;
+			Ok(())
+		})?;
```

Comment:
The error should be checked and covered before `transfer` and `set_lock`. When there is an error we should make sure we haven't changed the storage.

Reply:
Refactored the code, execution order got changed.

But, extrinsics behaviour is, All the modified storage states 
are restored back if there is any error in transactions.